UIforETW/ChildProcess.cpp

```diff
@@ -112,10 +111,10 @@ DWORD ChildProcess::ListenerThread()
 	return 0;
 }
 
-
+_Pre_satisfies_(!(this->hProcess_))
 bool ChildProcess::Run(bool showCommand, std::wstring args)
 {
-	assert(!hProcess_);
+	ATLASSERT(!hProcess_);

```

Comment:
> Just a thought.

It's a good one. I agree. We should probably also have a macro that `ASSERT(false)`s on conditions that should never be reached, e.g. `UIETW_NOTREACHED()` so that we're explicit.

As a side note: you mentioned (somewhere else) reentrancy issues with `assert`, and I can easily guess/imagine what they are (calling `abort()`, and then not actually aborting seems bad, doesn't it?), but am not *exactly* sure what you were referring to.

Can you elaborate?